Sources/StreamChat/Repositories/ConnectionRepository.swift

```diff
@@ -42,6 +42,10 @@ class ConnectionRepository {
         self.timerType = timerType
     }
 
+    func initialize() {
+        webSocketClient?.initialize()
+    }
```

Comment:
This needs to be introduced. Otherwise, we won't report the status of the new connection to the delegates. This is especially important when using ChatClient as a singleton. This is the problem:

1. connect | connectionStatus == .initialized
2. timout | connectinStatus == .disconnected
3. connect again | connectionStatus == .disconnected
4. Since we don't reset the `connectionStatus`, it keeps at the `.disconnected`, and the `.connecting` state is only reported after the token refreshing process.